src/Avalonia.Base/Threading/Dispatcher.Invoke.cs

```diff
@@ -627,14 +627,20 @@ public void Post(SendOrPostCallback action, object? arg, DispatcherPriority prio
     }
 
     /// <summary>
-    /// Returns a task awaitable that would invoke continuation on specified dispatcher priority
+    /// Returns a task that would invoke continuation on specified dispatcher priority
     /// </summary>
-    public DispatcherPriorityAwaitable AwaitWithPriority(Task task, DispatcherPriority priority) =>
-        new(this, task, priority);
+    public async Task AwaitWithPriority(Task task, DispatcherPriority priority)
+    {
+        await Resume(priority);
+        await task;
+    }
     
     /// <summary>
-    /// Returns a task awaitable that would invoke continuation on specified dispatcher priority
+    /// Returns a task that would invoke continuation on specified dispatcher priority
     /// </summary>
-    public DispatcherPriorityAwaitable<T> AwaitWithPriority<T>(Task<T> task, DispatcherPriority priority) =>
-        new(this, task, priority);
+    public async Task<T> AwaitWithPriority<T>(Task<T> task, DispatcherPriority priority)
```

Comment:
Breaking change (return value), also perf degradation since it now returns a task

Reply:
My intention was to reduce the number of types, but I agree with you and will revert it for `AwaitWithPriority`. Would it be possible to remove `AwaitWithPriority` later, in v12, since it can be rewritten using `Resume` after this PR? Then you can shrink the public API a bit and help the user get the more efficient code since an awaiter returned by `Resume` can be integrated into a compiler generated state machine for user's methods without any additional cost.

Reply:
Sorry for taking so long to come back. I had no energy sometime or simply to time at all.

So, the problem I see currently is that the current behavior is buggy too. If you tell at which priority the task should be awaited and if it's completed then you might continue with a wrong priority. If it's the desired behavior to continue execution if the task is already completed.